firebase-functions/functions/datacite.js

```diff
@@ -184,3 +185,567 @@ exports.getDoiStatus = functions.https.onCall(async (data) => {
+});
+
+exports.getDoiStatusPR78 = functions.https.onCall(async (data) => {
+
+  let prefix;
+  let authHash
+
+  functions.logger.log(data);
+
+  try {
+    prefix = (await admin.database().ref('admin').child(data.region).child("dataciteCredentials").child("prefix").once("value")).val();
+  } catch (error) {
+      console.error(`Error fetching Datacite Prefix for region ${data.region}:`, error);
+      return null;
+  }
+
+  try {
+    authHash = (await admin.database().ref('admin').child(data.region).child("dataciteCredentials").child("dataciteHash").once("value")).val();
+  } catch (error) {
+      console.error(`Error fetching Datacite Auth Hash for region ${data.region}:`, error);
+      return null;
+  } 
+
+  try {
+    const url = `${baseUrl}${data.doi}/`;
+    // TODO: limit response to just the state field. elasticsearch query syntax?
+    const response = await axios.get(url, {
+      headers: {
+        'Authorization': `Basic ${authHash}`
+      },
+    });
+    return response.data.data.attributes.state;
+  } catch (err) {
+    // if the error is a 401, throw a HttpsError with the code 'unauthenticated'
+    if (err.response && err.response.status === 401) {
+      throw new functions.https.HttpsError(
+        'unauthenticated',
+        'Error from DataCite API: Unauthorized. Please check your API credentials.'
+      );
+    }
+    // if the error is a 404, throw a HttpsError with the code 'not-found'
+    if (err.response && err.response.status === 404) {
+      if (data.doi.startsWith(`${prefix}/`)) {
+        return 'not found'
+      }
+      return 'unknown'
+    }
+    // initialize a default error message
+    let errMessage = 'An error occurred while fetching the DOI.';
+
+    // if there is an error response from DataCite, include the status and statusText from the API error
+    // if the error doesn't have a response, include the error message
+    if (err.response) {
+      errMessage = `from DataCite API: ${err.response.status} - ${err.response.statusText}`;
+    } else if (err.message) {
+      errMessage = err.message;
+    }
+
+    // throw a default HttpsError with the code 'unknown' and the error message
+    throw new functions.https.HttpsError('unknown', errMessage);
+  }
+
+});
+
+exports.getCredentialsStored = functions.https.onCall(async (data) => {
+  try {
+    const credentialsRef = admin.database().ref('admin').child(data).child("dataciteCredentials");
+    const authHashSnapshot = await credentialsRef.child("dataciteHash").once("value");
+    const prefixSnapshot = await credentialsRef.child("prefix").once("value");
+
+    const authHash = authHashSnapshot.val();
+    const prefix = prefixSnapshot.val();
+
+    // Check for non-null and non-empty
+    return authHash && authHash !== "" && prefix && prefix !== "";
+  } catch (error) {
+    console.error("Error checking Datacite credentials:", error);
+    return false;
+  }
+});
+
+exports.recordToDataCite = functions.https.onCall(async (data) => {
```

Comment:
the datacite prefix isn't sensitive informaton so recordToDataCite dosn't need to be a firebase function. In order to keep it consistent with the other templates I Think it's better for this to be in the front end with the other templates files as it was.